qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -526,25 +353,25 @@ def top_measurement(self, value: str) -> None:
         self._top_measurement = value
 
     @property
-    def circuit(self) -> Optional[QuantumCircuit]:
-        """ return circuit """
-        return self._circuit
-
-    @circuit.setter
-    def circuit(self, value: QuantumCircuit) -> None:
-        """ set circuit """
-        self._circuit = value
-
-    @property
-    def assignment(self) -> List[int]:
+    def assignment(self) -> Any:
```

Comment:
Should we say here that the return type is defined by the post_processing as per the problem. I guess the post processing can be problem specific which is why its defined in the problem rather than in the algorithm.

Reply:
Yeah i can put that in the docstring 🙂